packages/flutter_tools/lib/src/base/fingerprint.dart

```diff
@@ -168,8 +167,9 @@ class Fingerprint {
 
   @override
   // Ignore map entries here to avoid becoming inconsistent with equals
-  // due to differences in map entry order.
-  int get hashCode => hash2(_properties.length, _checksums.length);
+  // due to differences in map entry order. This is a really bad hash
+  // function and should eventually be deprecated and removed.
+  int get hashCode => _properties.length + _checksums.length;
```

Comment:
Not sure the implementation of `hash2()`, but this seems worse?

Reply:
The hashcode itself is kind of pointless anyway - since it is hashing the lengths of the checksum/properties.

Reply:
that's true